app/src/main/java/org/mifos/mobile/ui/update_password/UpdatePasswordScreen.kt

```diff
@@ -0,0 +1,253 @@
+                },
+                label = R.string.confirm_password,
+                supportingText = confirmPasswordErrorContent,
+                icon = R.drawable.ic_lock_black_24dp,
+                trailingIcon = {
+                    var image = if (viewConfirmPassword) Icons.Filled.Visibility
+                    else Icons.Filled.VisibilityOff
+
+                    if (!confirmPasswordError) {
+                        IconButton(onClick = { viewConfirmPassword = !viewConfirmPassword }) {
+                            Icon(
+                                imageVector = image,
+                                contentDescription = "password visibility Button"
+                            )
+                        }
+                    } else {
+                        Icon(imageVector = Icons.Filled.Error, contentDescription = null)
+                    }
+                },
+                error = confirmPasswordError,
+                visualTransformation = if (viewConfirmPassword) VisualTransformation.None else PasswordVisualTransformation(),
+                keyboardType = KeyboardType.Password
+
+            )
+
+            Button(
+                onClick = {
+                    keyboardController?.hide()
+                    newPasswordErrorContent = viewModel.getNewPasswordError(newPassword)
+                    confirmPasswordErrorContent = viewModel.getConfirmPasswordError(confirmPassword)
+
+                    when {
+                        newPasswordErrorContent.isEmpty() && confirmPasswordErrorContent.isEmpty() -> {
+                            viewModel.validateAndUpdatePassword(
+                                newPassword,
+                                confirmPassword
+                            )
+                        }
+
+                        newPasswordErrorContent.isEmpty() && confirmPasswordErrorContent.isNotEmpty() -> {
+                            confirmPasswordError = true
+                        }
+
+                        newPasswordErrorContent.isNotEmpty() && confirmPasswordErrorContent.isEmpty() -> {
+                            newPasswordError = true
+                        }
+
+                        else -> {
+                            confirmPasswordError = true
+                            newPasswordError = true
+                        }
+                    }
+                },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 16.dp, end = 16.dp, top = 4.dp),
+                contentPadding = PaddingValues(12.dp),
+                colors = ButtonDefaults.buttonColors(
+                    containerColor = if (isSystemInDarkTheme()) DarkPrimary
+                    else LightPrimary
+                )
+            ) {
+                Text(text = stringResource(id = R.string.change_password))
+            }
+        }
+    }
+}
+
+@Composable
+@Preview(showSystemUi = true)
+fun PreviewUpdatePasswordScreen(
```

Comment:
Convention is Name of the screen + Preview. So rename it to UpdatePasswordScreenPreview